dataactbroker/handlers/accountHandler.py

```diff
@@ -137,6 +144,117 @@ def login(self,session):
         return self.response
 
 
+    def max_login(self,session):
+        """
+
+        Logs a user in if their password matches
+
+        arguments:
+
+        session  -- (Session) object from flask
+
+        return the reponse object
+
+        """
+        try:
+            safeDictionary = RequestDictionary(self.request)
+
+            # Obtain POST content
+            ticket = safeDictionary.getValue("ticket")
+            service = safeDictionary.getValue('service')
+            parent_group = CONFIG_BROKER['parent_group']
+
+            # Call MAX's serviceValidate endpoint and retrieve the response
+            url = CONFIG_BROKER['cas_service_url'].format(ticket, service)
+            max_xml = requests.get(url).content
+            max_dict = xmltodict.parse(max_xml)
+
+            if not 'cas:authenticationSuccess' in max_dict['cas:serviceResponse']:
+                raise ValueError("You have failed to login successfully with MAX")
+
+            # Grab the email and list of groups from MAX's response
+            email = max_dict['cas:serviceResponse']['cas:authenticationSuccess']['cas:attributes']['maxAttribute:Email-Address']
+            group_list_all = max_dict['cas:serviceResponse']['cas:authenticationSuccess']['cas:attributes']['maxAttribute:GroupList'].split(',')
+            group_list = list(filter(lambda g:  g.startswith(parent_group), group_list_all))
+
+            # Deny access if not in the parent group aka they're not allowed to access the website all together
+            if not parent_group in group_list:
+                raise ValueError("You have logged in with MAX but do not have permission to access the broker. Please "
+                                 "contact DATABroker@fiscal.treasury.gov to obtain access.")
+
+            cgac_group = list(filter(lambda g: len(g) == len(parent_group + "-CGAC_")+3, group_list))
```

Comment:
Same question here about whether or not a list comprehension would be easier to read here.